### PR TITLE
[MAINTENANCE] bump python minimum version to 3.8

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
# What does this PR do? 
* Update CI/CD pipelines to use 3.8 as a minimum. 
* Update message on docs site 
* Remove deprecated code (from v0.14), mostly around BigQuery dependency.

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted

    ```
    black .

    ruff . --fix
    ```
- [x] Appropriate tests and docs have been updated

### CODE REVIEW

**Review by Kilo59 — CHANGES_REQUESTED** [review]
Very excited about this 😍 !!

https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#python-requirement

Need to update `setup.py` `python_requires` to say `>=3.8`.

Is this planned for next week's release?

File: ci/azure-pipelines.yml
Comment (Kilo59) [inline ci/azure-pipelines.yml:1]: Did you update the airflow constraints tests?

**Comment by Shinnnyshinshin** [comment]
Yes, @Kilo59 that's the current plan :), thanks for catching the change in `setup.py` adding now.

File: great_expectations/core/async_executor.py [hunk 1]
```diff
@@ -130,13 +133,13 @@ def patch_https_connection_pool(concurrency_config: ConcurrencyConfig) -> None:
     ###
     # NOTE: 20210907 - jdimatteo: The python requests pool size can bottleneck concurrency and result in warnings
     # like "WARNING  urllib3.connectionpool:connectionpool.py:304 Connection pool is full, discarding connection:
-    # bigquery.googleapis.com". To remove this bottleneck, patch the https connection pool as described at
-    # https://stackoverflow.com/a/22253656/1007353. After upgrading from the deprecated packages pybigquery and
-    # google-cloud-python to python-bigquery-sqlalchemy and python-bigquery, this patching code can be replaced
```
Comment (Shinnnyshinshin) [inline great_expectations/core/async_executor.py:135]: This was updated according to the instructions left by the contributor.

File: CONTRIBUTING_CODE.md [hunk 2]
```diff
@@ -2,25 +2,35 @@
...
-
-To discuss your code change before you implement it, join the [Great Expectations Slack community](https://greatexpectations.io/slack) and make your suggestion in the [#contributing](https://greatexpectationstalk.slack.com/archives/CV828B2UX) channel. 
+- [Contribute a code change](#contribute-a-code-change)
+  - [Prerequisites](#prerequisites)
+  - [Fork and clone the Great Expectations repository](#fork-and-clone-the-great-expectations-repository)
+  - [Install Python dependencies](#install-python-dependencies)
+  - [Create a virtual environment (optional)](#create-a-virtual-environment-optional)
+    - [Python](#python)
+    - [Anaconda](#anaconda)
+  - [Install dependencies from requirements-dev.txt](#install-dependencies-from-requirements-devtxt)
```
Comment (Shinnnyshinshin) [inline CONTRIBUTING_CODE.md]: Looks like this was re-generated.

File: great_expectations/expectations/core/expect_column_values_to_be_of_type.py [hunk 3]
```diff
@@ -571,10 +540,10 @@ def _get_dialect_type_module(
 
     # Bigquery works with newer versions, but use a patch if we had to define bigquery_types_tuple
     try:
-        if (
+        if BigQueryDialect and (
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/core/expect_column_values_to_be_of_type.py:543]: This `if BigQueryDialect` check is necessary

**Review by Kilo59 — APPROVED** [review]

File: CONTRIBUTING_CODE.md (hunk 2 above)
Comment (anthonyburdi) [reply CONTRIBUTING_CODE.md]: This looks out of sync with develop, is this maybe a merge conflict resolution issue? https://github.com/great-expectations/great_expectations/blob/develop/CONTRIBUTING_CODE.md

**Review by anthonyburdi — APPROVED** [review]
Awesome work! Love all the red! Approving but please just double check CONTRIBUTING_CODE.md with develop (aside from the python version changes).

**Pull request merged**